packages/shorebird_cli/lib/src/commands/release_new/release_new_command.dart

```diff
@@ -0,0 +1,420 @@
+  ///
+  /// Expectations for methods invoked by this command:
+  ///  - They perform their own logging. If an error occurs, they are
+  ///    responsible for properly logging the error, cleaning up running
+  ///    [Progress]es, etc.
+  ///  - They handle their own exceptions and exit with a non-zero exit code if
+  ///    an error occurs *instead of* throwing an exception.
+  @visibleForTesting
+  Future<void> createRelease(Releaser releaser) async {
+    await releaser.assertPreconditions();
+    await releaser.assertArgsAreValid();
+
+    // This command handles logging, we don't need to provide our own
+    // progress, error logs, etc.
+    final app = await codePushClientWrapper.getApp(appId: appId);
+    final targetFlutterRevision = await resolveTargetFlutterRevision();
+    try {
+      await shorebirdFlutter.installRevision(revision: targetFlutterRevision);
+    } catch (_) {
+      exit(ExitCode.software.code);
+    }
+
+    final releaseFlutterShorebirdEnv = shorebirdEnv.copyWith(
+      flutterRevisionOverride: targetFlutterRevision,
+    );
+    return await runScoped(
+      () async {
+        final releaseArtifact = await releaser.buildReleaseArtifacts();
+        final releaseVersion = await releaser.getReleaseVersion(
+          releaseArtifactRoot: releaseArtifact,
+        );
+
+        // Ensure we can create a release from what we've built.
+        await assertVersionIsReleasable(
+          version: releaseVersion,
+          flutterRevision: targetFlutterRevision,
+          releasePlatform: releaser.releaseType.releasePlatform,
+        );
+
+        // Ask the user to proceed (this is skipped when running via CI).
+        await confirmCreateRelease(
+          app: app,
+          releaseVersion: releaseVersion,
+          flutterVersion: targetFlutterRevision,
+          releasePlatform: releaser.releaseType.releasePlatform,
+        );
+
+        final release = await getOrCreateRelease(
+          version: releaseVersion,
+          releasePlatform: releaser.releaseType.releasePlatform,
+        );
+        await prepareRelease(release: release, pipeline: releaser);
+        await releaser.uploadReleaseArtifacts(release: release, appId: appId);
+        await finalizeRelease(release: release, pipeline: releaser);
+
+        logger
+          ..success('✅ Published Release ${release.version}!')
+          ..info(releaser.postReleaseInstructions);
+
+        printPatchInstructions(
+          releaser: releaser,
+          releaseVersion: releaseVersion,
+          releaseType: releaser.releaseType,
+          flavor: flavor,
+          target: target,
+        );
+      },
+      values: {
+        shorebirdEnvRef.overrideWith(() => releaseFlutterShorebirdEnv),
+      },
+    );
+  }
+
+  /// Determines which Flutter version to use for the release. This will be
+  /// either the version specified by the user or the version provided by
+  /// [shorebirdEnv]. Will exit with code 70 if the version specified by the
```

Comment:
Nit: should we do `[ExitCodes.software.code]` instead?

Reply:
Fixed